src/Resources/contao/classes/BackendUser.php

```diff
@@ -357,7 +360,10 @@ protected function setUserFromDb()
 		}
 
 		$GLOBALS['TL_USERNAME'] = $this->username;
-		$GLOBALS['TL_LANGUAGE'] = str_replace('_', '-', $this->language);
+		$GLOBALS['TL_LANGUAGE'] = str_replace('_', '-', $this->language); // backwards compatibility
+
+		$kernel->getContainer()->get('request_stack')->getCurrentRequest()->setLocale($this->language);
+		$kernel->getContainer()->get('translator')->setLocale($this->language);
```

Comment:
But only since we moved the exception templates. But Twig was pre-configured before, and it makes sense to make the translator work at all if we provide a full-stack app.